### DISCUSSION TITLE
Smart meter power direction issue (only positive values)

### DISCUSSION THREAD

**Discussion by KG3RK3N** [arendst/Tasmota#15869] [Q&A] [answered]
Hello, I have a problem with reading a power meter with SML and the direction (always positive values, no negative).
I have a Honeywell AS2020 smart power meter. The reading with a script and SML is already working, but I have no idea how I can read the current power with the right direction.

My current script:

```
>D
>B
->sensor53 r

>M 1
+1,3,s,0,9600,,1
1,77070100600100ff@#,Server-ID,,server_id,0
1,77070100010800ff@1000,Total Consumed,kWh,total_consumed_kwh,1
1,77070100020800ff@1000,Total Delivered,kWh,total_delivered_kwh,1
1,77070100100700ff@0.1,Current Consumption,W,current_power,0

#
```

Here is the difference from the dump, the value 62/63 defines the direction of the power.
```
77 07 01 00 10 07 00 ff 01 01 62 1e 52 ff 62 15 01 => -21W
77 07 01 00 10 07 00 ff 01 01 62 1e 52 ff 63 02 e3 01 => 739W
```

Has anyone an idea how I can get the direction and correct the output value?

**Comment by Jason2866** [comment, marked as answer]
Related PR #15886
